### PR TITLE
Fix Mul canonical to match that of Pow canonical

### PR DESCRIPTION
**Pull request by bocklund**
Test case from #1899

### CODE REVIEW

File: symengine/tests/basic/test_subs.cpp [hunk 1]
```diff
@@ -197,6 +197,14 @@ TEST_CASE("Mul: subs", "[subs]")
...
     REQUIRE(eq(*r1->subs(d), *Nan));
+
+    d.clear();
+    r1 = mul(real_double(2.0), x);
+    // xreplace with an empty mapping dict should be a no-op
+    r2 = r1->xreplace(d);
+    std::cout << "r1: " << *r1 << std::endl;
+    std::cout << "r2: " << *r2 << std::endl;
+    REQUIRE(eq(*r1, *r2));
 }
```
Comment (bocklund) [inline symengine/tests/basic/test_subs.cpp:208]: The failure also appears to affect `subs`. Does it make sense to add tests for both or only one or the other? 

Note that `rational` and `integer` types _do not_ fail.

**Comment by rikardn** [comment]
Good work! Should the new rules for canonicalization be documented in `mul.h`?

**Comment by isuruf** [comment]
The rule is already there. https://github.com/symengine/symengine/blob/562c95748070bb0d8d84df56c76566e4a63ec043/symengine/mul.h#L51-L54

**Review by rikardn — APPROVED** [review]
Ok, then I have no comments. This looks good.

**Comment by isuruf** [comment]
Thanks for the test @bocklund and for the review @rikardn

**Pull request merged**